packages/pigeon/platform_tests/test_plugin/android/src/main/kotlin/com/example/test_plugin/ProxyApiTestApiImpls.kt

```diff
@@ -0,0 +1,699 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package com.example.test_plugin
+
+import androidx.annotation.RequiresApi
+import io.flutter.plugin.common.BinaryMessenger
+
+class ProxyApiTestClass : ProxyApiSuperClass(), ProxyApiInterface
+
+open class ProxyApiSuperClass
+
+interface ProxyApiInterface
+
+@RequiresApi(25) class ClassWithApiRequirement
```

Comment:
Does pigeon have a different minimum api level from Flutter? I know this is a test but was surprised to see the annotation and I dont see where we protect against this classes reference on api 24 and below.

Reply:
This was added only to ensure that wrapping a class with an API requirement doesn't break the generated code or cause lint warnings. There is an example implementation that protects against it in `ProxyApiTestApiImpls.kt`: https://github.com/flutter/packages/pull/6371/files#diff-87909fbd5b2dd9a1652c6dfd584c2ff3f387a4d4a8f4a00e32e553b6aacb6272R691.